### PR TITLE
Using Franklin Form block

### PR DESCRIPTION
**Pull request by jalagari**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--airlessco--hlxsites.hlx.page/na/en/contactus/
- After: https://main--airlessco--hlxsites.hlx.page/na/en/contactus/new

### CODE REVIEW

File: blocks/form/form.js [hunk 1]
```diff
@@ -1,9 +1,49 @@
...
+  if (value) {
+    element.setAttribute(key, value);
+  }
+}
+
+function setConstraints(fd, element) {
+  if (fd.Mandatory === 'true') {
+    element.setAttribute('required', true);
+  }
+  setElementProps(element, 'pattern', fd.pattern);
```
Comment (vdua) [inline blocks/form/form.js:22]:
```
setElementProps(element, entries) {
  entries.filter(([p, v]) => v).forEach(([p,v]) => element.setAttribute(p, v))
}
```
and use it
```
setElementProps(element, Object.entries({
   'pattern' : fd.Pattern,
   'min' : fd.Min,
   'max' : fd.Max,
   'minlength' : fd.Min,
   'maxlength' : fd.Max
}))
```

Comment (jalagari) [reply blocks/form/form.js:22]: Min & Max only need to set on input type number & date for input type string, email, etc. need to set minlength & maxlength

**Review by bstopp — CHANGES_REQUESTED** [review]
This PR breaks the functionality and visual appearance of the current implementation.

I was of the understanding that there would be no regressions with the introduction of a new form block.

**Comment by jalagari** [comment]
> This PR breaks the functionality and visual appearance of the current implementation.
> 
> I was of the understanding that there would be no regressions with the introduction of a new form block.

@bstopp its not breaking any functionality or visual appearance. If you are looking at  https://main--airlessco--hlxsites.hlx.page/na/en/contactus/new then it doesn't have my changes. 

Can you please provide me access to create branch so that I can check-in in branch and then you can check behaviour and apperance of form.

**Comment by bstopp** [comment]
Unfortunately, i don't have that level of access.

**Comment by bstopp** [comment]
I merged this change into a branch, there are some discrepancies between the forms. Here's a screenshot of both on the same page:

screencapture-feat-improve-form-airlessco-hlxsites-hlx-page-na-en-contactus-2023-02-27-10_05_35

It also looks like only the English version of the form was provided?

**Comment by bstopp** [comment]
Here's the branch I am using to verify. If you push more changes here, i'll be able to merge and re-check.

https://github.com/hlxsites/airlessco/tree/feat/improve-form

**Pull request closed without merging**